ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -105,8 +104,8 @@ class IndexSegment implements Iterable<IndexEntry> {
   private int indexSizeExcludingEntries;
   private int firstKeyRelativeOffset;
   private IFilter bloomFilter = null;
-  private int valueSize;
-  private int persistedEntrySize;
+  private int valueSize = VALUE_SIZE_INVALID_VALUE;
+  private int maxEntrySize = 0;
```

Comment:
why max value 0 ?

Reply:
It's the initial value. When we have the first entry, will set the entry size of the first entry. Then increase if there is larger entry.